diff --git a/mesh/packet_muxer.go b/mesh/packet_muxer.go
index 242eee1..23abaf9 100644
--- a/mesh/packet_muxer.go
+++ b/mesh/packet_muxer.go
@@ -2,12 +2,13 @@ package mesh
 
 import (
 	"fmt"
+	"strings"
+	"time"
+
 	"github.com/evilsocket/islazy/async"
 	"github.com/evilsocket/islazy/log"
 	"github.com/google/gopacket"
 	"github.com/google/gopacket/pcap"
-	"strings"
-	"time"
 )
 
 const (
@@ -54,9 +55,9 @@ func NewPacketMuxer(iface, filter string, workers int) (mux *PacketMuxer, err er
 		}
 		defer inactiveHandle.CleanUp()
 
-		if err = inactiveHandle.SetRFMon(true); err != nil {
-			log.Warning("error while setting interface %s in monitor mode: %s", iface, err)
-		}
+		// if err = inactiveHandle.SetRFMon(true); err != nil {
+		// 	log.Warning("error while setting interface %s in monitor mode: %s", iface, err)
+		// }
 
 		if err = inactiveHandle.SetSnapLen(SnapLength); err != nil {
 			return nil, fmt.Errorf("error while settng span len: %s", err)
