diff --git a/mesh/packet_muxer.go b/mesh/packet_muxer.go
index 242eee1..79ff9ec 100644
--- a/mesh/packet_muxer.go
+++ b/mesh/packet_muxer.go
@@ -2,12 +2,13 @@ package mesh
 
 import (
 	"fmt"
+	"strings"
+	"time"
+
 	"github.com/evilsocket/islazy/async"
 	"github.com/evilsocket/islazy/log"
 	"github.com/google/gopacket"
 	"github.com/google/gopacket/pcap"
-	"strings"
-	"time"
 )
 
 const (
@@ -54,8 +55,12 @@ func NewPacketMuxer(iface, filter string, workers int) (mux *PacketMuxer, err er
 		}
 		defer inactiveHandle.CleanUp()
 
-		if err = inactiveHandle.SetRFMon(true); err != nil {
-			log.Warning("error while setting interface %s in monitor mode: %s", iface, err)
+		if strings.Contains(iface, "mon") {
+			if err = inactiveHandle.SetRFMon(true); err != nil {
+				log.Warning("error while setting interface %s in monitor mode: %s", iface, err)
+			}
+		} else {
+			log.Info("interface %s dont contains 'mon' and should be already in monitor mode")
 		}
 
 		if err = inactiveHandle.SetSnapLen(SnapLength); err != nil {
